### PR TITLE
Fix FAB while scrolling on Dashboard

### LINKED ISSUE #145: Hide Floating action button on scroll
The FAB can hide the text behind it.
Make it so that it hides when scrolling down and comes back up when scrolling up.

Files concerned: 
* tripslist.dart
* dashboard.dart

### PR DESCRIPTION
**Pull request by rish07**
Fixes #145 

Please let me know if there are any changes @kshitijalwadhi

### CODE REVIEW

**Comment by Deepanshu-Rohilla** [comment]
Hey @rish07 . Please attach some GIF's to demonstrate the changes (similar to screenshots in your previous PR).

**Review by kshitijalwadhi — CHANGES_REQUESTED** [review]
@rish07 
Please ensure that your hardwrap settings is set at 900. There seems to be some extra changes in the files changed. 
Also, execute `flutter analyze` before submitting your pull request and fix the bugs it suggests. You current PR is failing that check and suggesting this:
`Omit type annotations for local variables • lib/screens/tripslist.dart:26:5 • omit_local_variable_types`

**Comment by rish07** [comment]
@kshitijalwadhi will fix the flutter analyze bug but I can't seem to find the hard wrapping option in my Android Studio:

image

**Comment by kshitijalwadhi** [comment]
You are at the correct place afaik. Try changing that 80 to 900 and save the files again.

**Comment by rish07** [comment]
@Deepanshu-Rohilla 
ezgif com-gif-maker

**Comment by rish07** [comment]
@kshitijalwadhi Changed it to 900 and fixed the analyze bug.

**Comment by kshitijalwadhi** [comment]
The problem with hard-wrap still seems to be there.
@Deepanshu-Rohilla can you guide him how to fix that in Android Studio? 
I have only worked on VS Code.

**Comment by rish07** [comment]
@kshitijalwadhi one way to fix it is to merge this PR, pull the new code on your system, change the formatting and push the changes as an amend to my previous commit

**Comment by Deepanshu-Rohilla** [comment]
Screenshot 2020-12-12 at 20 32 17
@rish07 This is the setting in my machine currently (macOS). 
Make sure that after changing the settings from 80 to 900, you've again reformatted the changed files (using `reformat code with dartfmt` option).
If the problem still persists, you check https://github.com/devclub-iitd/ShareACab/blob/master/CONTRIBUTING.md for stackoverflow links.

**Comment by rish07** [comment]
image

@Deepanshu-Rohilla There doesn't seem to be any such option in my version of the Android Studio(4.1.1).

**Review by kshitijalwadhi — CHANGES_REQUESTED** [review]
There are some issues with the way you have implemented this. The scrolling is jittery and it keeps bringing me back to the top when scrolling up. Have a look at the GIF attached. Maybe you will have to look more into its implementation.

**Comment by rish07** [comment]
Ah right @kshitijalwadhi just saw it on my physical device. Let me find another approach to this. The problem is FAB is on Dashboard widget but the scrolling in the triplist.dart. So even tho I can detect the scroll direction in the child, the only way I can update the state of the parent is to call a function that sets the state of the Dashboard widget. That is what I have done here, I am not sure why is it setting states abruptly. Do you have any approach in mind?

File: lib/screens/dashboard.dart [hunk 1]
```diff
@@ -34,6 +36,10 @@ class _DashboardState extends State<Dashboard> with AutomaticKeepAliveClientMixi
     setState(() {});
   }
 
+  void refreshState() {
+    setState(() {});
+  }
+
```
Comment (Deepanshu-Rohilla) [inline lib/screens/dashboard.dart:42]: Calling refreshState makes the screen go to top. Make sure you don't use it whenever someone scrolls down.

**Comment by rish07** [comment]
@kshitijalwadhi fixed the hard wrapping issue

File: lib/screens/tripslist.dart [hunk 2]
```diff
@@ -19,10 +23,26 @@ class _TripsListState extends State<TripsList> {
...
         builder: (_, usersnapshot) {
+          _scrollController
+            ..addListener(() {
+              if (_scrollController.position.userScrollDirection.toString() !=
+                  globals.scrollDirection) {
+                setState(() {
+                  globals.scrollDirection =
+                      _scrollController.position.userScrollDirection.toString();
+                });
+                widget.refreshParent();
```
Comment (rish07) [inline lib/screens/tripslist.dart:37]: @Deepanshu-Rohilla it is only being called when the direction of the scroll changes

File: lib/screens/dashboard.dart [hunk 3]
```diff
@@ -124,45 +136,58 @@ class _DashboardState extends State<Dashboard> with AutomaticKeepAliveClientMixi
...
-                                Navigator.push(context, MaterialPageRoute(builder: (context) => GroupPage()));
-                              },
-                              icon: Icon(Icons.group),
-                              label: Text('Group'),
-                            ),
-                          )
-                    : null,
+                floatingActionButtonLocation:
+                    FloatingActionButtonLocation.endFloat,
+                floatingActionButton: globals.scrollDirection ==
```
Comment (rish07) [inline lib/screens/dashboard.dart]: And if I don't call set state it won't get the new value of globals.scrollDirection

**Comment by rish07** [comment]
I would suggest moving the FAB on triplist.dart, so we can access setState only when needed. Or do let me know if there's any other approch

**Comment by Deepanshu-Rohilla** [comment]
Try out the approach that you think might work out.

**Comment by tmibvishal** [comment]
Closing this pull request now. Because this feature was already added in another pr.

**Pull request closed without merging**